gtsfm/loader/loader_base.py

```diff
@@ -81,6 +95,73 @@ def is_valid_pair(self, idx1: int, idx2: int) -> bool:
             validation result.
         """
 
+    def get_image(self, index: int) -> Image:
+        """Get the image at the given index, satisfying a maximum image resolution constraint.
+
+        Determine how the camera intrinsics and images should be jointly rescaled based on desired img. resolution.
+        Each loader implementation should set a `_max_resolution` attribute.
+
+        Args:
+            index: the index to fetch.
+
+        Raises:
+            IndexError: if an out-of-bounds image index is requested.
+
+        Returns:
+            Image: the image at the query index. It will be resized to satisfy the maximum
+                allowed loader image resolution if the full-resolution images for a dataset
+                are too large.
+        """
+        # no downsampling may be required, in which case target_h and target_w will be identical
+        # to the full res height & width.
+        img_full_res = self.get_image_full_res(index)
+        (
+            _,
+            _,
+            target_h,
+            target_w,
+        ) = img_utils.get_downsampling_factor_per_axis(img_full_res.height, img_full_res.width, self._max_resolution)
+        logger.info(
```

Comment:
Do we need to log this? I feel we do not need to, and if you disagree, can we please make it `debug` so that we can filter it out?

Reply:
I found it pretty useful so far to me to remember what resolution we're working at. If it's too verbose in the CI and in future runs, I'll remove it : - )

Reply:
maybe we should log it just for the first 5 images?